### PR TITLE
Centralize releasability checks

### PR DESCRIPTION
**Pull request by dprotaso**
Going to do a follow up on the documentation in another PR

Related: Dropping the template that's synced across repos here https://github.com/knative-sandbox/knobots/pull/285

### CODE REVIEW

**Comment by kvmware** [comment]
I do not like the multiple files for each repo. This would be much less overhead to have all the logic in one file that runs through a matrix of repos (external file that one can add/remove from). That would eliminate the multitude of files and the boiler plate script.

File: .github/workflows/releasability.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# Copyright 2023 The Knative Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'releasability'
+
+on:
+  workflow_call:
+    inputs:
+      repo:
```
Comment (kvmware) [inline .github/workflows/releasability.yaml:9]: The repos here should just be read in from the `repos.yaml` file and matrix across this workflow.

**Review by kvmware — CHANGES_REQUESTED** [review]
I can look into the badges of docs but I don't know if having badges on the docs site is worth this.

**Comment by dprotaso** [comment]
Badge are only at the workflow level

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

**Comment by dprotaso** [comment]
Also the badges aren't on the docs site but the documentation here - https://github.com/knative/release/blob/main/TIMELINE.md

**Comment by kvmware** [comment]
I created an issue here https://github.com/knative/release/issues/193 that outlines the concerns here that can be taken care of at another time. For now the good done with this change out weighs the concerns.

**Pull request merged**